Additional nested configuration option causes gpio module to not start anymore

When adding the configuration option `digital:` the gpio module does not start anymore.

### Configuration
```yaml
  ### Interrupt Debounce
  #debounce: 50 # in milliseconds

  ### Single Pin Configuration
  my-pin:

    ### Pin Number
    pin: 5

    ### Pin Name
    # Subsitutions:
    # - {pin} = pin number
    # - {pin:02d} = pin number padded with 0's to always be 2 numbers
    # - {pin_id} = name of this section
    #name: '{pin_id}'

    ### Pin Mode
    pin mode: 'output' # can be 'input' or 'output'
    digital:
    initial state: OFF

    ### Pin Resistor
    #resistor: 'off' # can be 'pullup', 'pulldown', or 'off'

    ### Pin Type Specifc Options
    # Each pin type will have some options specific to it,
    # consult the documentation for details.

  ### Multiple Pin Configuration
```

### Logs
```text
2020-12-06 08:20:50,306 [INFO ] [core                    ] MQTTany 0.12.0 starting
2020-12-06 08:20:50,550 [INFO ] [core.gpio               ] Detected board: RASPBERRY_PI_3B
2020-12-06 08:20:50,861 [ERROR] [gpio                    ] An exception occurred while running function 'load'
2020-12-06 08:20:50,863 [ERROR] [gpio                    ]

Traceback (most recent call last):
  File "/opt/mqttany/mqttany/common.py", line 319, in _call
    retval = func(**kwargs)
  File "/opt/mqttany/mqttany/modules/gpio/core.py", line 94, in load
    config_data = parse_config(config_raw, conf_options, log)
  File "/opt/mqttany/mqttany/config.py", line 256, in parse_config
    valid, config = parse_dict(data, options)
  File "/opt/mqttany/mqttany/config.py", line 245, in parse_dict
    config,
  File "/opt/mqttany/mqttany/config.py", line 139, in process_option
    section_valid, section_config = parse_dict(value, option)
  File "/opt/mqttany/mqttany/config.py", line 249, in parse_dict
    key, data.pop(key, "**NO DATA**"), options[key], config
  File "/opt/mqttany/mqttany/config.py", line 165, in process_option
    if not isinstance(value, option["type"]):
TypeError: isinstance() arg 2 must be a type or tuple of types

2020-12-06 08:20:50,864 [WARN ] [core                    ] Module 'gpio' load failed
2020-12-06 08:20:50,871 [INFO ] [core                    ] Module 'mqtt' started successfully
2020-12-06 08:20:50,897 [INFO ] [mqtt                    ] Connected to broker '192.168.2.211:1883'
```

After removing the `digital:` config option the module loads as expected:

```text
2020-12-06 08:23:53,857 [INFO ] [core                    ] MQTTany 0.12.0 starting
2020-12-06 08:23:54,100 [INFO ] [core.gpio               ] Detected board: RASPBERRY_PI_3B
2020-12-06 08:23:54,444 [INFO ] [core                    ] Module 'mqtt' started successfully
2020-12-06 08:23:54,452 [INFO ] [core                    ] Module 'gpio' started successfully
2020-12-06 08:23:54,462 [INFO ] [gpio.digital            ] Setting up 'my-pin' on GPIO05 as OUTPUT
2020-12-06 08:23:54,477 [INFO ] [mqtt                    ] Connected to broker '192.168.2.211:1883'
```

This issue was discussed here:

https://community.openhab.org/t/mqttany-gpio-and-more-over-mqtt/81450/8?u=sihui

So with the valid indentation (which I notice you did have in the forum, with the other error) I get the following error in the log:

```log
2020-12-06 17:27:15,621 [ERROR] [core    ] [gpio                    ] Value 'False' for config option 'initial state' is not one of ['ON', 'on', 'OFF', 'off']
2020-12-06 17:27:15,622 [WARN ] [core    ] [gpio                    ] Optional section 'digital' is not valid, it will be ignored
```

which explains the error you see farther on in the log saying that the `digital` key is missing:

```log
2020-12-06 17:27:15,668 [ERROR] [core    ] [gpio                    ] An exception occurred while running function 'load'
2020-12-06 17:27:15,672 [ERROR] [core    ] [gpio                    ] 

Traceback (most recent call last):
  File "/opt/mqttany/mqttany/common.py", line 319, in _call
    retval = func(**kwargs)
  File "/opt/mqttany/mqttany/modules/gpio/core.py", line 116, in load
    pin_object = build_pin(id, named_config)
  File "/opt/mqttany/mqttany/modules/gpio/core.py", line 83, in build_pin
    clazz = clazz(pin, CONFIG[CONF_KEY_MODE], id, name, pin_config)
  File "/opt/mqttany/mqttany/modules/gpio/pin/digital.py", line 119, in __init__
    self._edge = pin_config[CONF_KEY_DIGITAL][CONF_KEY_INTERRUPT]
KeyError: 'digital'

```

The config parser incorrectly reports that the configuration is valid, even though the `digital` section has an invalid value. This partly makes sense, because the `digital` section is a conditional section and the code handling that is related to the optional section code. I need to change the handling of invalid values in sections with conditions that are true.

I will get a patch done up shortly to fix this.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,9 @@
 
 ## Development
 
+* **Changed**
+  * Config parser no long converts selection keys to strings before matching.
+
 * **Added**
   * Config parser now considers required sections with defaults for all options to be
     valid and will populate the section in the returned configuration.
@@ -9,6 +12,8 @@
 * **Fixed**
   * Config parser was reporting valid configuration for optional conditional sections
     containing invalid values. Fixes problem 1 in [#80](https://github.com/CrazyIvan359/mqttany/issues/80).
+  * YAML returning booleans for unquoted `on` or `off` strings, booleans have been
+    added as accepted values where applicable to avoid confusion. Fixes [#80](https://github.com/CrazyIvan359/mqttany/issues/80).
 
 ## 0.12.0
 

--- a/mqttany/config.py
+++ b/mqttany/config.py
@@ -184,16 +184,16 @@ def parse_config(
                                 return False
 
                         elif "selection" in option:
-                            if str(value) in option["selection"]:
+                            if value in option["selection"]:
                                 log.trace(
                                     "Got selection '%s' for config option '%s'",
                                     value,
                                     name,
                                 )
                                 if isinstance(option["selection"], dict):
-                                    config[name] = option["selection"][str(value)]
+                                    config[name] = option["selection"][value]
                                 else:
-                                    config[name] = str(value)
+                                    config[name] = value
                             else:
                                 log.error(
                                     "Value '%s' for config option '%s' is not one of %s",

--- a/mqttany/modules/gpio/pin/digital.py
+++ b/mqttany/modules/gpio/pin/digital.py
@@ -68,6 +68,7 @@ CONF_OPTIONS = {
                 "pulldown": PinBias.PULL_DOWN,
                 "down": PinBias.PULL_DOWN,
                 "off": PinBias.NONE,
+                False: PinBias.NONE,
                 "none": PinBias.NONE,
             },
         },
@@ -88,7 +89,14 @@ CONF_OPTIONS = {
             },
             CONF_KEY_INVERT: {"type": bool, "default": False},
             CONF_KEY_INITIAL: {
-                "selection": {"ON": True, "on": True, "OFF": False, "off": False},
+                "selection": {
+                    "ON": True,
+                    "on": True,
+                    True: True,
+                    "OFF": False,
+                    "off": False,
+                    False: False,
+                },
                 "default": False,
             },
         },

--- a/mqttany/modules/i2c/device/mcp230xx.py
+++ b/mqttany/modules/i2c/device/mcp230xx.py
@@ -96,6 +96,7 @@ CONF_OPTIONS = {
                                 "pullup": Resistor.PULL_UP,
                                 "up": Resistor.PULL_UP,
                                 "off": Resistor.OFF,
+                                False: Resistor.OFF,
                                 "none": Resistor.OFF,
                             },
                         },
@@ -104,8 +105,10 @@ CONF_OPTIONS = {
                             "selection": {
                                 "ON": True,
                                 "on": True,
+                                True: True,
                                 "OFF": False,
                                 "off": False,
+                                False: False,
                             },
                             "default": False,
                         },

